docs/src/reference.md

```diff
@@ -0,0 +1,121 @@
+# Reference
+
+The API reference of `NNlib`.
+
+## Activation Functions
+
+Non-linearities that go between layers of your model. Note that, unless otherwise stated, activation functions operate on scalars. To apply them to an array you can call `σ.(xs)`, `relu.(xs)` and so on.
+
+```@docs
+celu
+elu
+gelu
+hardsigmoid
+sigmoid_fast
+hardtanh
+tanh_fast
+leakyrelu
+lisht
+logcosh
+logsigmoid
+mish
+relu
+relu6
+rrelu
+selu
+sigmoid
+softplus
+softshrink
+softsign
+swish
+hardswish
+tanhshrink
+trelu
+```
+
+## Softmax
+
+```@docs
+softmax
+logsoftmax 
+```
+
+## Pooling
+
+```@docs
+PoolDims
+maxpool
+meanpool
+```
+
+## Padding
+
+```@docs
+pad_reflect
+pad_constant
+pad_repeat
+pad_zeros
+```
+
+## Convolution
+
+```@docs
+conv
+ConvDims
+depthwiseconv
+DepthwiseConvDims
+DenseConvDims
+```
+
+## Upsampling
+
+```@docs
+upsample_nearest
```

Comment:
For many of these sections, it might be worth pointing briefly to what Flux layers use these functions?

Reply:
Should it be something like this -
```
## Softmax

\```@docs
softmax
logsoftmax
\```

### Relevant Flux functionality

\```@docs
Flux.xyz
\```
```

Or should I simply point it out in text -

`NNlib.softmax` is used by `Flux.xyz`(Redirect to the reference in Flux's documentation, can't use `@ref` as `Flux.xyz` is not present in `NNlib`'s docs) ...

Reply:
Maybe it suffices to summarize at the top of section:
````
## Upsampling

Used by `Flux.x`, `Flux.y`, ...

```@docs
upsample_nearest
```
````

Reply:
Yes something like this was what I pictured. "There's an Upsample layer in Flux for which this is the backend" 

I don't think we should repeat the Flux docstrings here, too much duplication. This API page might end up included entire in Flux's docs, too.